tokio/Cargo.toml

```diff
@@ -107,8 +104,8 @@ mio = { version = "0.8.6", optional = true, default-features = false }
 num_cpus = { version = "1.8.0", optional = true }
```

Comment:
Could this not be replaced by `std::thread::available_parallelism`?

Reply:
Well, this is outside the scope of this PR.

This PR is for bumping MSRV for tokio only, you can open another PR/issue/discussion for this.

Reply:
available_parallelism was stabilized with an incomplete implementation, and this is expected to cause regressions (e.g., OOM) in Rust 1.63 and smaller that don't have cgroupv1 support (https://github.com/rust-lang/rust/pull/97925).

So I think we need to stick to num_cpus until MSRV became 1.64+.